tensorflow-framework/src/main/java/org/tensorflow/framework/utils/ShapeUtils.java

```diff
@@ -63,64 +63,36 @@
-        Operand<TInt64> ldims = (Operand<TInt64>) dims;
-        ldims.asOutput().data().scalars().forEach(s -> result.add(s.getLong()));
-      } else if (dType.equals(TUint8.DTYPE)) {
-        @SuppressWarnings("unchecked")
-        Operand<TUint8> udims = (Operand<TUint8>) dims;
-        udims.asOutput().data().scalars().forEach(s -> result.add(s.getObject().longValue()));
-      } else { // shouldn't happen
-        throw new IllegalArgumentException("the data type must be an integer type");
-      }
-
-    } else {
-      try (Session session = new Session((Graph) scope.env())) {
-        if (dType.equals(TInt32.DTYPE)) {
-          try (Tensor<TInt32> tensorResult =
-              session.runner().fetch(dims).run().get(0).expect(TInt32.DTYPE)) {
-            tensorResult.data().scalars().forEach(s -> result.add((long) s.getInt()));
-          }
-        } else if (dType.equals(TInt64.DTYPE)) {
-          try (Tensor<TInt64> tensorResult =
-              session.runner().fetch(dims).run().get(0).expect(TInt64.DTYPE)) {
-            tensorResult.data().scalars().forEach(s -> result.add(s.getLong()));
-          }
-        } else if (dType.equals(TUint8.DTYPE)) {
-          try (Tensor<TUint8> tensorResult =
-              session.runner().fetch(dims).run().get(0).expect(TUint8.DTYPE)) {
-            tensorResult.data().scalars().forEach(s -> result.add(s.getObject().longValue()));
-          }
-        } else { // shouldn't happen
-          throw new IllegalArgumentException("the data type must be an integer type");
-        }
-      }
+      return getLongArray(dims.asTensor());
+    }
+    try (Session session = new Session((Graph)scope.env());
+        Tensor tensor = session.runner().fetch(dims).run().get(0)) {
+      return getLongArray(tensor);
     }
-    return result.stream().mapToLong(i -> i).toArray();
   }
 
   /**
-   * Gets the shape for the data within a Tensor
+   * Converts a TInt32 or TInt64 to a java long array
    *
-   * @param tensor the tensor
-   * @return the Shape of the tensor's data;
+   * @param scope the TensorFlow scope
+   * @param dims the dimension tensor
+   * @param <T> the type of the dimensions, must either be TInt32 or TInt64 type
+   * @return the long array
+   * @throws java.lang.IllegalArgumentException if the dims type is not an integer
    */
-  public static <T extends TNumber> Shape getShape(Tensor<T> tensor) {
-    NdArray<?> data = (NdArray<?>) tensor.data();
-    return data.shape();
+  public static long[] getLongArray(Tensor dims) {
+    List<Long> result = new ArrayList<>();
+    if (dims instanceof TInt32) {
+      ((TInt32)dims).scalars().forEach(s -> result.add((long) s.getInt()));
+    } else if (dims instanceof TInt64) {
+      ((TInt64)dims).scalars().forEach(s -> result.add(s.getLong()));
+    } else if (dims instanceof TUint8) {
+      ((TUint8)dims).scalars().forEach(s -> result.add(s.getObject().longValue()));
+      } else { // shouldn't happen
```

Comment:
Formatting seems off in this else block. Also maybe we should have a `TIntegral` type to go with `TFloating` and then the else block really can't happen. And when we get sealed types in a future Java version we can have the compiler check it for us.

Reply:
`TIntegral` is also planned, I was keeping this for another PR as well, as there is more functions in the framework that will need to be updated.